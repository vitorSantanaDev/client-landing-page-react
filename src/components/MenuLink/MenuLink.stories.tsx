import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'
import MenuLink from '.'

export default {
  title: 'MenuLink',
  component: MenuLink,
  args: {
    children: 'My Link',
    link: 'https://google.com.br'
  }
} as ComponentMeta<typeof MenuLink>

export const Default: ComponentStory<typeof MenuLink> = (args) => (
  <div style={{ display: 'flex' }}>
    <MenuLink {...args} />
  </div>
)
