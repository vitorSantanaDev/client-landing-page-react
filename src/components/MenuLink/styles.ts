import styled, { css } from 'styled-components'

export const Wrapper = styled.a`
  ${({ theme }) => css`
    display: block;
    text-decoration: none;
    position: relative;

    padding: ${theme.spacings.small};
    color: ${theme.colors.primaryColor};
    font-size: ${theme.font.sizes.small};

    &::after {
      content: '';
      width: 0%;
      height: 0.2rem;

      bottom: 0;
      left: 50%;
      position: absolute;

      transition: all 300ms ease-in-out;
      background-color: ${theme.colors.secondaryColor};
    }

    &:hover::after {
      left: 25%;
      width: 50%;
    }
  `}
`
