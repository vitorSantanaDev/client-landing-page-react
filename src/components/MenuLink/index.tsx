import React from 'react'
import * as S from './styles'

interface MenuLinkProps {
  link: string
  newTab?: boolean
}

const MenuLink: React.FC<MenuLinkProps> = ({
  children,
  link,
  newTab = false
}) => {
  const target = newTab ? '_blank' : '_self'
  return (
    <S.Wrapper href={link} target={target}>
      {children}
    </S.Wrapper>
  )
}

export default MenuLink
