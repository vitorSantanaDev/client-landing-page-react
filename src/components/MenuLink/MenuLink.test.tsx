import React from 'react'
import { screen } from '@testing-library/react'
import { renderTheme } from '../../styles/theme/render-theme'
import MenuLink from '.'

describe('<MenuLink />', () => {
  it('should render a link', () => {
    renderTheme(<MenuLink link="https://google.com">My Link</MenuLink>)
    expect(screen.getByRole('link', { name: /my link/i })).toHaveAttribute(
      'target',
      '_self'
    )
  })
  it('should render open in a new tab', () => {
    renderTheme(
      <MenuLink link="https://google.com" newTab>
        My Link
      </MenuLink>
    )
    expect(screen.getByRole('link', { name: /my link/i })).toHaveAttribute(
      'target',
      '_blank'
    )
  })

  it('should match snapshot', () => {
    const { container } = renderTheme(
      <MenuLink link="https://google.com" newTab>
        My Link
      </MenuLink>
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
