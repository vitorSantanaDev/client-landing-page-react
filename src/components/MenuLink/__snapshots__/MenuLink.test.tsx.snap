// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuLink /> should match snapshot 1`] = `
.c0 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
  padding: 1.6rem;
  color: #0A1128;
  font-size: 1.6rem;
}

.c0::after {
  content: '';
  width: 0%;
  height: 0.2rem;
  bottom: 0;
  left: 50%;
  position: absolute;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  background-color: #dc143c;
}

.c0:hover::after {
  left: 25%;
  width: 50%;
}

<a
  class="c0"
  href="https://google.com"
  target="_blank"
>
  My Link
</a>
`;
